rewrite-xml/src/main/java/org/openrewrite/xml/internal/Namespaces.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2024 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.xml.internal;
+
+import lombok.Value;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+@Value
+public class Namespaces {
+
+    Map<String, String> namespaces = new HashMap<>();
```

Comment:
A mutable map here might lead to problems with our immutability convention and detecting changes.